Update CCM docs for new tag normalization feature (#32407)
DataDog/documentation: content/en/cloud_cost_management/tags/_index.md

- Tag values up to 5000 characters are supported

Cloud Cost Recommendations use [standard Datadog metrics normalization rules][14]. Tag values in recommendations are converted to lowercase and limited to 200 characters.
For example, a tag `Team:Engineering-Services` would appear as `team:engineering-services` in recommendations, but as `team:Engineering-Services` in cost data. 

## Override tag value normalization

> Turn on **Tag Normalization** in the Tag Pipelines page to normalize all cost tag values to match the Metrics normalization. From the example above, you would see `team:engineering-services` everywhere. The Tag Normalization toggle is enabled by default, with normalized tag values backfilled for the past 3 months automatically, for all new customers. To backfill normalized tags for a longer period up to 15 months, contact [Datadog support][13].

Reviewer: ```suggestion
Turn on **Tag Normalization** in the Tag Pipelines page to normalize all cost tag values to match the Metrics normalization. From the example above, you would see `team:engineering-services` everywhere. For all new users, the Tag Normalization toggle is enabled by default, with normalized tag values backfilled for the past 3 months automatically. To backfill normalized tags for a longer period up to 15 months, contact [Datadog support][13].
```

Tag normalization allows you to:
- View, filter and group Cost Recommendations and cost data with the same tag values
- Correlate cost and observability data with the same tag values

> - Automatically combine tag values with different casing (ex: `env:Engineering-Prod` and `env:engineering-Prod` will both be normalized to `env:engineering-prod` without having to manually set up Tag Pipelines)

Reviewer: ```suggestion
- Automatically combine tag values with different casing (For example: `env:Engineering-Prod` and `env:engineering-Prod` are both normalized to `env:engineering-prod` without having to manually set up Tag Pipelines)
```
